udpTester.py

```diff
@@ -358,47 +384,47 @@ def receiver(args):
     print(
         f"Received {totalMsgs} complete messages out of {expectedCount}, lost {lost:.1f}%"
     )
+    print(f"Received {duplicatePackets} duplicate packets")
 
 
 def create_parser():
     # shared args
     usageString = """
-    Running the transmitter:
-        udpTester.py transmitter address outgoing [-h] [-p PORT] [-m MESSAGESIZE] [-t TOTALCOUNT] [-s PACKETSIZE] [-i INTERVAL] [-l LOSSINESS]
+    udpTester.py -transmitter -a ADDRESS -o OUTGOING [-h] [-p PORT] [-t TOTALCOUNT] [-m MESSAGESIZE] [-s PACKETSIZE] [-i INTERVAL] [-l LOSSINESS]
 
-    Running the receiver:
-        udpTester.py receiver address [-h] [-p PORT] [-m MESSAGESIZE] [-t TOTALCOUNT] [-s PACKETSIZE] [-r REPORTINTERVAL] [-b RECEIVEBUFFER] [-q QUIET]
+    udpTester.py -receiver -a ADDRESS [-h] [-p PORT] [-t TOTALCOUNT] [-m MESSAGESIZE] [-s PACKETSIZE] [-b RECEIVEBUFFER] [-r REPORTINTERVAL] [-q QUIET]
 
-    Note that outgoing is a mandatory argument for the transmitter. It is the network interface address used for outgoing packets.
+    Example:
+        python3 udpTester.py -transmitter -a 239.0.0.1 -o 192.168.2.8 -t 200 -m 450 -s 150
+        python3 udpTester.py -receiver -a 239.0.0.1 -t 200 -m 450 -s 150 -b 200000
     """
     parser = argparse.ArgumentParser(
         usage=usageString, description="Test multicast traffic"
     )
     parser.add_argument(
-        "role", help="The role selected", type=str, choices=["transmitter", "receiver"]
-    )
-    parser.add_argument(
-        "address",
-        help="The ip address to use. Multicast addresses range from 224.0.0.0 to 239.255.255.255",
+        "-a",
+        "--address",
+        help="Mandatory argument: The ip address to use. Multicast addresses range from 224.0.0.0 to 239.255.255.255",
         type=str,
```

Comment:
If mandatory, why not pass `required=True` as keyword argument? Saves you the null check down below. Instead of type=str you can also pass `type=ip_validator` and then you never need to validate that ip address in the actual code, see [here](https://docs.python.org/3/library/argparse.html#type)

Reply:
I think I did that initially. If you put the required keyword argument, it will give a warning when you don't provide it. @kurtuluso suggested to simply print the usage after a wrong input.